javascript-source/systems/commercialSystem.js

```diff
@@ -0,0 +1,173 @@
+                    /**
+                    * @commandpath commercial autotimer (interval_mins) (length_secs) [message] - Sets the autotimer
+                    */
+                    } else {
+                        var valid_lengths = ["30", "60", "90", "120", "150", "180"];
+                        if (args.length < 3 || isNaN(args[1]) || isNaN(args[2]) || args[1] < 8 || !valid_lengths.includes(args[2])) {
+                            $.say($.whisperPrefix(sender) + $.lang.get('commercialsystem.autotimer.bad-parm'));
+                            return;
+                        }
+                        
+                        argsString = args.slice(3).join(' ');
+                        $.inidb.set('commercialSettings', 'length', args[2]);
+                        $.inidb.set('commercialSettings', 'interval', args[1]);
+                        $.inidb.set('commercialSettings', 'message', argsString);
+                        $.inidb.set('commercialSettings', 'commercialtimer', true.toString());
+                        
+                        commercialLength = parseInt(args[2]);
+                        commercialInterval = parseInt(args[1]);
+                        commercialMessage = argsString;
+                        commercialTimer = true;
+                        
+                        startCommercialTimer();
+                        
+                        $.say($.whisperPrefix(sender) + $.lang.get('commercialsystem.autotimer.status-on', commercialLength, commercialInterval));
+                        if (commercialMessage.length() > 0) {
+                            $.say($.whisperPrefix(sender) + $.lang.get('commercialsystem.autotimer.status-on-msg', commercialMessage));
+                        } else {
+                            $.say($.whisperPrefix(sender) + $.lang.get('commercialsystem.autotimer.status-on-nomsg'));
+                        }
+                    }
+                    return;
+                }
+            }
+
+            /**
+             * @commandpath commercial (length) [silent] - Runs a commercial, optionally does not post a success message to chat
+             */
+            if (args.length >= 1 && !isNaN(args[0])) {
+                var result = $.twitch.RunCommercial($.channelName, args[0]);
+                
+                if (result.getInt("_http") === 422) {
+                    $.say($.whisperPrefix(sender) + $.lang.get('commercialsystem.422'));
+                } else {
+                    lastCommercial = $.systemTime();
+                    
+                    if (args.length < 2 || !args[1].equalsIgnoreCase("silent")) {
+                        $.say($.whisperPrefix(sender) + $.lang.get('commercialsystem.run', args[0]));
+                    }
+                }
+            }
+        }
+    });
+
+    // Set the interval to run commercials
+    if (commercialTimer) {
```

Comment:
Would you mind moving this to the `initReady` event, then it would only start the timer if the module is enabled, or gets enabled.